prov/gni/src/gnix_ep.c

```diff
@@ -423,15 +425,71 @@ ssize_t gnix_ep_tsenddata(struct fid_ep *ep, const void *buf, size_t len,
  * Base EP API function implementations.
  ******************************************************************************/
 
+
+static int gnix_ep_control(fid_t fid, int command, void *arg)
+{
+	int i, ret = FI_SUCCESS;
+	struct gnix_fid_ep *ep;
+	struct gnix_fid_domain *dom;
+	struct gnix_vc *vc;
+
+	ep = container_of(fid, struct gnix_fid_ep, ep_fid);
+
+	switch (command) {
+	/*
+	 * for FI_EP_RDM, post wc datagrams now
+	 */
+	case FI_ENABLE:
+		if (ep->type == FI_EP_RDM) {
+			dom = ep->domain;
+			for (i = 0;
+			     i < dom->fabric->n_wc_dgrams; i++) {
+				ret = _gnix_vc_alloc(ep,
+				       FI_ADDR_UNSPEC, &vc);
+				if (ret != FI_SUCCESS)
+					GNIX_WARN(FI_LOG_EP_CTRL,
+				     "_gnix_vc_alloc call returned %d\n",
+				     ret);
+				if (ret == FI_SUCCESS) {
+					ret = _gnix_vc_accept(vc);
+					if (ret != FI_SUCCESS) {
+						GNIX_WARN(FI_LOG_EP_CTRL,
+					    "_gnix_vc_accept call returned %d\n",
+						     ret);
+						_gnix_vc_destroy(vc);
+						goto err;
+					} else {
+						dlist_insert_tail(&vc->entry,
+					  &ep->wc_vc_list);
+				}
+			}
+		}
+		}
```

Comment:
I'm confused about what this curly brace is closing.  Am I mis-counting?  In any case, the indentation is confusing.